app/drivers/kscan/kscan_gpio_matrix.c

```diff
@@ -262,19 +272,21 @@ static int kscan_gpio_config_interrupts(const struct device **devices,
                 LOG_ERR("Unable to find output GPIO device");                                      \
                 return -EINVAL;                                                                    \
             }                                                                                      \
-            err = gpio_pin_configure(output_devices[o], out_cfg->pin,                              \
-                                     GPIO_OUTPUT_ACTIVE | out_cfg->flags);                         \
+            err =                                                                                  \
+                gpio_pin_configure(output_devices[o], out_cfg->pin, GPIO_OUTPUT | out_cfg->flags); \
             if (err) {                                                                             \
                 LOG_ERR("Unable to configure pin %d on %s for output", out_cfg->pin,               \
                         out_cfg->label);                                                           \
                 return err;                                                                        \
             }                                                                                      \
         }                                                                                          \
         data->dev = dev;                                                                           \
-        COND_CODE_1(CONFIG_ZMK_KSCAN_MATRIX_POLLING,                                               \
-                    (k_timer_init(&data->poll_timer, kscan_gpio_timer_handler, NULL);), ())        \
         (COND_CODE_0(DT_INST_PROP(n, debounce_period), (k_work_init), (k_delayed_work_init)))(     \
             &data->work, kscan_gpio_work_handler_##n);                                             \
+        COND_POLL_OR_INTERRUPTS(                                                                   \
+            (k_timer_init(&data->poll_timer, kscan_gpio_timer_handler_##n, NULL);                  \
+             kscan_gpio_set_output_state_##n(dev, 0);),                                            \
+            (kscan_gpio_set_output_state_##n(dev, 1);))                                            \
```

Comment:
Would I be correct in guessing that the interrupt behavior is different here because it sets all the outputs active, waits for an interrupt to happen on any input, and then does a regular scan?

Reply:
Correct. We want to drive all outputs to active when "idle", so that any key press will trigger an interrupt. Then we keep polling/scanning until all keys are released again.